packages/core/core/src/BundleGraph.js

```diff
@@ -544,6 +544,27 @@ export default class BundleGraph {
       return null;
     }
 
+    if (
```

Comment:
This addresses the issue, but adding more special casing to `getDependencyResolution` feels less than ideal. We should consider rethinking our approach to resolving dependencies.